src/main/java/zmq/SocketBase.java

```diff
@@ -1135,10 +1135,16 @@ final void startReaping(Poller poller)
 
         //  Initialize the termination and check whether it can be deallocated
         //  immediately.
+        setCtxTerminated();
         terminate();
         checkDestroy();
     }
 
+	private final void setCtxTerminated()
+    {
+        ctxTerminated = true;
+    }
+
```

Comment:
I don't think this is sufficient in solving the deadlock. It's still not atomically setting the boolean value. Consider making ctxTerminated a volatile or changing it to be AtomicBoolean. I'd prefer the latter it's easier to make correct.

Reply:
You are right, and new codes were committed just now. Thanks for your reply.